src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -12668,12 +12668,28 @@ namespace SurfaceGeometry {
             }
             for (J = 1; J <= M; ++J) {
                 Ind = SurfCollinearVerts(J);
-                for (K = Ind; K <= NSides - J; ++K) {
+                if (Ind > NSides) {
+                    Ind = Ind - NSides + M - 1;
+                }
+                for (K = Ind; K <= NSides - 1; ++K) {
                     SurfaceTmp(SurfNum).Vertex(K - J + 1).x = SurfaceTmp(SurfNum).Vertex(K - J + 2).x;
                     SurfaceTmp(SurfNum).Vertex(K - J + 1).y = SurfaceTmp(SurfNum).Vertex(K - J + 2).y;
                     SurfaceTmp(SurfNum).Vertex(K - J + 1).z = SurfaceTmp(SurfNum).Vertex(K - J + 2).z;
                 }
             }
+            // remove duplicated points and resize Vertex
+            Array1D<Vector> OldVertex;
```

Comment:
This method is only called once per simulation, right?

Reply:
@Myoldmopar The method is called at each heat transfer surface with collinear points.